### ISSUE TITLE
Fix SBT deprecations

### ISSUE THREAD

**Issue by Ma27** [Ma27/rediscala#1]
```
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:83: `<++=` operator is deprecated. Try `lhs ++= { x.value }`
[warn]   or see http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html.
[warn]       scalacOptions in (Compile, doc) <++= baseDirectory in LocalProject("rediscala") map { bd =>
[warn]                                       ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:90: `<++=` operator is deprecated. Try `lhs ++= { x.value }`
[warn]   or see http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html.
[warn]       scalacOptions in (Compile, doc) <++= version in LocalProject("rediscala") map { version =>
[warn]                                       ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:97: `<++=` operator is deprecated. Try `lhs ++= { x.value }`
[warn]   or see http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html.
[warn]       siteMappings <++= (mappings in packageDoc in Compile, version in LocalProject("rediscala")) { (mm, version) =>
[warn]                    ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:102: `<<=` operator is deprecated. Use `key := { x.value }` or `key ~= (old => { newValue })`.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]       cleanSite <<= (updatedRepository, git.gitRunner, streams, version in LocalProject("rediscala")) map { (dir, git, s, v) =>
[warn]                 ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:111: `<<=` operator is deprecated. Use `key := { x.value }` or `key ~= (old => { newValue })`.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]       synchLocal <<= (cleanSite, privateMappings, updatedRepository, ghpagesNoJekyll, git.gitRunner, streams) map { (clean, mappings, repo, noJekyll, git, s) =>
[warn]                  ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:126: `<<=` operator is deprecated. Use `key := { x.value }` or `key ~= (old => { newValue })`.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]     sourceDirectory in BenchTest <<= baseDirectory / "src/benchmark",
[warn]                                  ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:131: `<<=` operator is deprecated. Use `key := { x.value }` or `key ~= (old => { newValue })`.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]     testGrouping in BenchTest <<= definedTests in BenchTest map partitionTests
[warn]                               ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:46: trait Build in package sbt is deprecated: Use .sbt format instead
[warn] object RediscalaBuild extends Build {
[warn]                               ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:50: value defaultSettings in object Defaults is deprecated: Default settings split into coreDefaultSettings and IvyModule/JvmModule plugins.
[warn]   lazy val standardSettings = Defaults.defaultSettings ++
[warn]                                        ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:97: method t2ToApp2 in object Scoped is deprecated: The sbt 0.10 style DSL is deprecated: '(k1, k2) map { (x, y) => ... }' should now be '{ val x = k1.value; val y = k2.value }'.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]       siteMappings <++= (mappings in packageDoc in Compile, version in LocalProject("rediscala")) { (mm, version) =>
[warn]                         ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:102: method t4ToTable4 in object Scoped is deprecated: The sbt 0.10 style DSL is deprecated: '(k1, k2) map { (x, y) => ... }' should now be '{ val x = k1.value; val y = k2.value }'.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]       cleanSite <<= (updatedRepository, git.gitRunner, streams, version in LocalProject("rediscala")) map { (dir, git, s, v) =>
[warn]                     ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:111: method t6ToTable6 in object Scoped is deprecated: The sbt 0.10 style DSL is deprecated: '(k1, k2) map { (x, y) => ... }' should now be '{ val x = k1.value; val y = k2.value }'.
[warn] See http://www.scala-sbt.org/0.13/docs/Migrating-from-sbt-012x.html
[warn]       synchLocal <<= (cleanSite, privateMappings, updatedRepository, ghpagesNoJekyll, git.gitRunner, streams) map { (clean, mappings, repo, noJekyll, git, s) =>
[warn]                      ^
[warn] /home/travis/build/Ma27/rediscala/project/Rediscala.scala:126: method / in class RichFileSetting is deprecated: Use a standard setting definition.
[warn]     sourceDirectory in BenchTest <<= baseDirectory / "src/benchmark",
[warn]                                                    ^
[warn] 13 warnings found
```

**Comment by herzrasen** [comment]
I'd prefer using a build.sbt where possible instead of Rediscala.scala. What do you think?

**Comment by Ma27** [comment]
agreed. A build.sbt is more convenient and less verbose. I’ll have a look at it when I’m home again :-)

**Comment by kardapoltsev** [comment]
If nobody starts working on it I could try to make PR with  a fixes. Has anybody already started?

**Comment by herzrasen** [comment]
I didn't start yet. Kind of busy right now. Feel free to open a PR @kardapoltsev

**Comment by Ma27** [comment]
@kardapoltsev not yet, I added you as second assignee, would be cool if you could submit a PR, @herzrasen or I could take over after some time :-)

**Comment by Ma27** [comment]
closed as of #4
